### DISCUSSION TITLE
Confusion about the interpolation from a small supercell to a denser q-mesh.

### DISCUSSION THREAD

**Discussion by jgSi1113** [SSCHAcode/python-sscha#346] [Q&A] [answered]
Dear SSCHA developers,

I noticed that the link [https://github.com/orgs/SSCHAcode/discussions/172](url) says we can extend the SSCHA minimisation of small supercell to a larger one. Such work may obtain better data with less HPC resources.  I confused about the start point (q-mesh) of harmonic calculation. 

I have finished the harmonic calculation with a 4*4*4 q-mesh, which shows imaginary phonon modes. I want to study the anharmonic effect on the system via SSCHA. If I use a 4*4*4 supercell, the calculations is too expensive for me. If I use a 2*2*2 q-mesh, the harmonic phonon is too bad. So I want to create a 2*2*2 supercell for SSCHA minimisation and then extend the results to 4*4*4 supercell.  Should I do a harmonic calculation with a 2*2*2 q-mesh as start points as a start points for the SSCHA calculations?  If not, how to start with the 4*4*4 meshed harmonic dynamical matrix to create a  2*2*2 supercell for the further SSCHA calculations?

Best & Regards!
Jianguo Si

**Comment by mesonepigreco** [comment, marked as answer]
Hi,

Yes, you can perform the SSCHA calculation on a 2x2x2 dynamical matrix, then try the interpolation on a 4x4x4 exploiting the Harmonic dynamical matrices computed in the 4x4x4.

To this purpouse:
```
# Load the final sscha dynamical matrix
dyn_sscha_222 = CC.Phonons.Phonons("my_sscha_222_", nqirr)

# Load the harmonic dynamical matrix in the 2x2x2 q-mesh
dyn_harm_222 =  CC.Phonons.Phonons("harmonic_222_", nqirr)

# Load the harmonic dynamical matrix in the 4x4x4 q-mesh
dyn_harm_444 =  CC.Phonons.Phonons("harmonic_444_", nqirr_big)

# Get the sscha dynamical matrix in a 4x4x4
dyn_sscha_444 = dyn_sscha_222.Interpolate( (2,2,2), (4,4,4), dyn_harm_222, dyn_harm_444)

# Save the final sscha dynamical matrix in a 4x4x4
dyn_sscha_444.save_qe("my_sscha_444_")
```
where ``nqirr`` and ``nqirr_big`` are the number of irreducible q-points for the 2x2x2 and 4x4x4 q-mesh, respectively (the number of dynamical matrices files).

Here, we are interpolating **the difference** between the SSCHA and the Harmonic dynamical matrix from the 2x2x2 into the 4x4x4 q-mesh, then summing this difference on the Harmonic dynamical matrix in the 4x4x4. This usually converge much faster than interpolating the SSCHA dynamical matrix, as the difference between SSCHA and Harmonic is shorter-range than the full dynamical matrix, thus converges on a smaller q-mesh.

 Hope this is usefull,
All the best,
LM

**Reply by jgSi1113** [reply]
Thanks for your kind reply. I will finish it with your workflow.

Thanks and Regards!
Jianguo

**Reply by biglinn** [reply]
Hi, @mesonepigreco:
```
# Load the final sscha dynamical matrix
dyn_sscha_222 = CC.Phonons.Phonons("my_sscha_222_", nqirr)

# Load the harmonic dynamical matrix in the 2x2x2 q-mesh
dyn_harm_222 =  CC.Phonons.Phonons("harmonic_222_", nqirr)

# Load the harmonic dynamical matrix in the 4x4x4 q-mesh
dyn_harm_444 =  CC.Phonons.Phonons("harmonic_444_", nqirr_big)

# Get the sscha dynamical matrix in a 4x4x4
dyn_sscha_444 = dyn_sscha_222.Interpolate( (2,2,2), (4,4,4), dyn_harm_222, dyn_harm_444)

# Save the final sscha dynamical matrix in a 4x4x4
dyn_sscha_444.save_qe("my_sscha_444_")
```
Is these 5 steps what you said `interpolate the difference of your last SSCHA dynamical matrix with the starting one`?

Does it mean:
If I have completed SSCHA iterations in 222 supercell and got the final converged dynamical matrices(222 q-mesh),  I can directly get converged 444 q-mesh dynamical matrices instead of do SSCHA iterations in 444 supercell.

**Reply by biglinn** [reply]
I have taken a try. It seems to get nice results having the same quality with 222 results.
However, does this method really consider the long-range interaction in 444 supercell?
Does the infomation of long-range come from the 444 harmonic dynamical matrices?  If so, could you introduce me this method's theory?
